Gemfile

```diff
@@ -1,5 +1,6 @@
 source 'https://rubygems.org'
 
 gem "activerecord", ">=7"
+gem "sqlite3", "~> 1.4"
```

Comment:
This is a constraint on ActiveRecord connection https://github.com/rails/rails/blob/v7.1.3.4/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L14
Without it, we will get higher version of sqlite3 which break the connection.
[Latest beta version](https://github.com/rails/rails/blob/v7.2.0.beta2/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L14) already fixed the problem, we just need to wait until the stable version to upgrade.